examples/chat-rooms-messages/react/src/app/page.tsx

```diff
@@ -0,0 +1,91 @@
+"use client";
+
+import { useRoom, useMessages, useChatClient } from '@ably/chat/react';
+import { Message } from '@ably/chat';
+import { useState } from 'react';
+import '../../styles/styles.css'
+
+export default function Home() {
+  const { roomStatus, connectionStatus } = useRoom();
+
+  if (roomStatus !== 'attached' || connectionStatus !== 'connected') {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div>
+      <Chat />
+    </div>
+  );
+}
+
+const Chat = () => {
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [message, setMessage] = useState('');
+  const { clientId } = useChatClient();
+  const { send } = useMessages({
+    listener: (message) => {
+      console.log('Received message: ', message);
+      setMessages((prevMessages: Message[]) => [...prevMessages, { ...message.message }]);
+    },
+  });
+
+  const handleSubmit = (e: React.FormEvent) => {
```

Comment:
Generally to avoid unnecessary re-renders, you'd use a `useCallback` here. Doing so would mean you need to change the current `useState` containing your message text to a `useRef`. If this is a simple demo, then perhaps it's not super important, so feel free to ignore this